schema/ingestion_config/v1.0.0/ingestion_config_models_extended.py

```diff
@@ -651,6 +663,29 @@ def valid_sources(cls: Self, source_list: List[AnnotationSource]) -> List[Annota
                 if has_glob_string and has_glob_strings:
                     multiple_glob_strings_errors.append((i, shape))
 
+        # For verifying that all source entries do not only have parent_filters or exclude
+        only_parent_filter_in_source_entry_errors = []
+        only_exclude_in_source_entry_errors = []
+
+        for i, source_element in enumerate(source_list):
+            has_parent_filters = False
+            has_exclude = False
+            has_shape = False
+            for shape in source_element.model_fields:
+                if getattr(source_element, shape) is None:
+                    continue
+                if shape == "exclude":
+                    has_exclude = True
+                elif shape == "parent_filters":
+                    has_parent_filters = True
+                else:
+                    has_shape = True
+
+            if not has_shape and has_parent_filters:
+                only_parent_filter_in_source_entry_errors.append(i)
+            if not has_shape and has_exclude:
+                only_exclude_in_source_entry_errors.append(i)
```

Comment:
Maybe we could have a constant for filter fields? 
Also, this could be combined with the iterations that are happening above? 
```suggestion
            for key_field in source_element.model_fields:
                if key_field not in {"parent_filters", "exclude"} and not getattr(source_element, shape):
                    continue            
                 has_shape = True
                 break

            if not has_shape: 
               total_errors.append(ValueError(f"Source entry {i} does not have a shape"))
               for filter_fields in {"parent_filters", "exclude"} :
                   if getattr(source_element, filter_fields):
                        total_errors.append(ValueError(f"Source entry {i} cannot have a {filter_fields} entry without a shape"))
```

Reply:
I was thinking that for the sake of clarity, we keep it separate?

Reply:
Not sure, if this is making it more clearer. But, this is not a blocker.

Reply:
Ok, fixed!